Docs (#616)
allenai/open-instruct: docs/algorithms/grpo.md

GRPO is an online RL method used in [DeepSeek R1 paper](https://arxiv.org/abs/2501.12948) and its first appearance is in [DeepSeekMath](https://arxiv.org/abs/2402.03300)

## Implemented Variants

- `grpo_fast.py` is a faster variant using [packing techniques](https://huggingface.co/blog/sirluk/llm-sequence-packing).

> - `grpo_vllm_thread_ray_gtrl.py` is the original GRPO implementation, using vLLM and Ray.

Reviewer: nit:
```suggestion
- `grpo_vllm_thread_ray_gtrl.py` is a more vanilla GRPO implementation, using vLLM and Ray.
```
